lib/middleware/request-handler.ts

```diff
@@ -406,11 +412,17 @@ export function elevateRequest (req: Interface.IOurRequest,
         res.write(text + JSON.stringify(data));
     };
 
-    res.sendEnd = (status: string, message: string): void => {
-        var text = util.format('%s%s,%s}',
+    res.sendEnd = (status: string,
```

Comment:
That will actually be kind of complicated, because the functions as currently written assume that you have zero or more calls to `sendChunk` followed immediately by a call to `sendEnd` with no other calls in between writing to the result. We may want to rethink how those functions work if we anticipate more of this sort of thing.

In the short run I might prefer the approach you have here (i.e. augmenting `sendEnd` with final data, although maybe more generically, e.g., supplying both key & value instead of assuming it's `correlationIds`) over trying to make `sendCorrelationIds` work, just for simplicity's sake.